.github/workflows/check.yaml

```diff
@@ -0,0 +1,36 @@
+name: Check
+
+on:
+    pull_request:
+        branches: ["**"]
+    push:
+        branches: ["main"]
+    merge_group:
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+env:
+    FORCE_COLOR: 3
+    TURBO_TOKEN: ${{ secrets.VERCEL_TECHNANCE_TOKEN }}
+    TURBO_TEAM: ${{ vars.TURBO_TEAM }}
+    NODE_VERSION: "22"
+    PNPM_VERSION: "10.6.5"
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - name: Setup
+              uses: technance-foundation/github-actions/setup@main
+              with:
+                  node-version: ${{ env.NODE_VERSION }}
+                  pnpm-version: ${{ env.PNPM_VERSION }}
+                  npm-token: ${{ secrets.NPM_TOKEN }}
+                  pnpm-cache: "read"
+                  build: false
+            - name: Test
+              shell: bash
+              run: "pnpm run test"
```

Comment:
Even though you're running the tests in the workspace root using Turbo, but the tests for `next-smoke` project is still not running, it's because the project is not part of the Pnpm workspace here.

Once https://github.com/technance-foundation/code-style/pull/42#discussion_r2507830683 is addressed, the tests should run properly across all packages and apps in the workspace.